cogs/birthday/birthday.py

```diff
@@ -512,52 +543,100 @@ async def setSelfBirthday(self, ctx: Context, *, birthday: MonthDayConverter):
         birthdayStr = "{0:%B} {0:%d}".format(birthday)
 
         if birthdayConfig:
-            confirmationStr = (
-                f"Are you sure you want to set your birthday to "
-                f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}? "
-                "Only administrators and moderators can reset your birthday afterwards. "
-                f"Type {bold('`yes`', escape_formatting=False)} to confirm."
+            confirmationStr = "\n".join(
+                (
+                    f"Are you sure you want to set your birthday to "
+                    f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}? "
+                    "Only administrators and moderators can reset your birthday afterwards.",
+                    f"Type {bold('`yes`', escape_formatting=False)} to confirm.",
+                )
             )
 
-            await ctx.author.send(f"{headerWarn}: {confirmationStr}")
+            async def mainFlow(destination, carefree: bool = False):
+                # define the time for the message to live for
+                SENSITIVE_MSG_TTL = None if carefree else 10.0
+                try:
+                    await destination.send(
+                        f"{headerWarn}: {confirmationStr}",
+                        delete_after=SENSITIVE_MSG_TTL,
+                    )
+                except discord.Forbidden:
+                    # catch failed DM
+                    raise
+                # wait for answer
+                responseChannel = None
+                if carefree:
+                    responseChannel = destination.dm_channel
+                else:
+                    responseChannel = destination.channel
```

Comment:
Can we have this as an argument for `mainFlow` instead? It doesn't look to me `carefree` means to differentiate, but is rather a privacy-sensitivity switch.
Also if this sounds right to you, do use `channel.send()` instead of `destination.send()`. This way we can sanity check that we are using 1 communication channel for messages we send *and* messages we expect back (e.g., we expect messages back when we do `check` functions). Doing `destination.send()` in one place, and then doing a check for `msg.channel == responseChannel` in another place gives an impression that we are possibly sending the message 1 way where the channel is implicitly decided by `discord.py` API, *and then* expect back a message from the channel we *explicitly* select. Practically the `Context.send` and `User.send` methods should forward content to the correct channels `Context.channel` and `User.dm_channel`, but better opt for a more explicit choice now so in the future when we come back to this code we won't have to think about this again.

Reply:
As per the comment above regarding the Union argument, I've changed it so that the only input that we need is the channel that this will be sent to and if the messages are carefree or not. 

All the checks and sends are dynamically selected based on the input `discord.TextChannel` input.